lib/util/parser.js

```diff
@@ -192,7 +192,7 @@ export function parse(html, supplemental, options = {}) {
     if (lastTextPos === -1 && matchOffset > 0) {
       let string = html.slice(0, matchOffset);
 
-      if (string && string.trim() && !doctypeEx.exec(string)) {
+      if (string && string.trim && !doctypeEx.exec(string)) {
```

Comment:
@tbranyen So this essentially means `if (isNotAllWhitespaceOnly && !doctypeEx.exec(string)) {`